File: ports/c#/test/Antlr4CodeCompletion.CoreUnitTest/Utils/CountingErrorListener.cs
using Antlr4.Runtime;
using Antlr4.Runtime.Misc;

namespace Antlr4CodeCompletion.CoreUnitTest.Utils
{
    /// <summary>
    /// </summary>
    /// <remarks>
    /// Port of antlr-c3 java unit test library to c#
    /// The c3 engine is able to provide code completion candidates useful for
    /// editors with ANTLR generated parsers, independent of the actual
    /// language/grammar used for the generation.
    /// https://github.com/mike-lischke/antlr4-c3
    /// </remarks>
    public class CountingErrorListener : BaseErrorListener
    {
        public int ErrorCount { get; set; } = 0;

        public override void SyntaxError([NotNull] IRecognizer recognizer, [Nullable] IToken offendingSymbol, int line, int charPositionInLine, [NotNull] string msg, [Nullable] RecognitionException e)
        {
            base.SyntaxError(recognizer, offendingSymbol, line, charPositionInLine, msg, e);
            this.ErrorCount++;
        }
    }
}

File: ports/java/src/test/java/com/vmware/antlr4c3/CountingErrorListener.java
package com.vmware.antlr4c3;

import org.antlr.v4.runtime.BaseErrorListener;
import org.antlr.v4.runtime.RecognitionException;
import org.antlr.v4.runtime.Recognizer;

public class CountingErrorListener extends BaseErrorListener {
    public int errorCount = 0;

    @Override
    public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol, int line, int charPositionInLine,
            String msg, RecognitionException e) {
        super.syntaxError(recognizer, offendingSymbol, line, charPositionInLine, msg, e);
        errorCount++;
    }
}
